app/src/ui/multi-commit-operation/choose-branch/rebase-choose-branch-dialog.tsx

```diff
@@ -0,0 +1,208 @@
+      </>
+    )
+  }
+
+  protected renderActionStatusIcon = () => {
+    return (
+      <ActionStatusIcon
+        status={this.rebasePreview}
+        classNamePrefix="merge-status"
+      />
+    )
+  }
+
+  protected updateStatus = async (baseBranch: Branch) => {
+    const { currentBranch: targetBranch, repository } = this.props
+    const computingRebaseForBranch = baseBranch.name
+
+    this.rebasePreview = {
+      kind: ComputedAction.Loading,
+    }
+    this.updateRebaseStatusPreview(baseBranch)
+
+    const { commits, base } = await promiseWithMinimumTimeout(async () => {
+      const commits = await getCommitsBetweenCommits(
+        repository,
+        baseBranch.tip.sha,
+        targetBranch.tip.sha
+      )
+
+      const base = await getMergeBase(
+        repository,
+        baseBranch.tip.sha,
+        targetBranch.tip.sha
+      )
+
+      return { commits, base }
+    }, 500)
+
+    // if the branch being track has changed since we started this work, abandon
+    // any further state updates (this function is re-entrant if the user is
+    // using the keyboard to quickly switch branches)
+    if (computingRebaseForBranch !== baseBranch.name) {
+      this.rebasePreview = null
+      this.updateRebaseStatusPreview(baseBranch)
+      return
+    }
+
+    // if we are unable to find any commits to rebase, indicate that we're
+    // unable to proceed with the rebase
+    if (commits === null) {
+      this.rebasePreview = {
+        kind: ComputedAction.Invalid,
+      }
+      this.updateRebaseStatusPreview(baseBranch)
+      return
+    }
+
+    // the target branch is a direct descendant of the base branch
+    // which means the target branch is already up to date and the commits
+    // do not need to be applied
+    const isDirectDescendant = base === baseBranch.tip.sha
+    const commitsOrIgnore = isDirectDescendant ? [] : commits
+
+    this.rebasePreview = {
+      kind: ComputedAction.Clean,
+      commits: commitsOrIgnore,
+    }
+    this.updateRebaseStatusPreview(baseBranch)
+  }
+
+  private updateRebaseStatusPreview(baseBranch: Branch) {
+    this.setState({ statusPreview: this.getRebaseStatusPreview(baseBranch) })
+  }
+
+  private getRebaseStatusPreview(baseBranch: Branch): JSX.Element | null {
+    if (this.rebasePreview == null) {
+      return null
+    }
+
+    const { currentBranch } = this.props
+
+    if (this.rebasePreview.kind === ComputedAction.Loading) {
+      return this.renderLoadingRebaseMessage()
+    }
+    if (this.rebasePreview.kind === ComputedAction.Clean) {
+      return this.renderCleanRebaseMessage(
+        currentBranch,
+        baseBranch,
+        this.rebasePreview.commits.length
+      )
+    }
+
+    if (this.rebasePreview.kind === ComputedAction.Invalid) {
+      return this.renderInvalidRebaseMessage()
+    }
+
+    return null
+  }
+
+  private renderLoadingRebaseMessage() {
+    return <>Checking for ability to rebase automatically...</>
```

Comment:
I know this was just copy&pasted, but maybe it's a good opportunity to replace those `...` with `…` 😳 
```suggestion
    return <>Checking for ability to rebase automatically…</>
```